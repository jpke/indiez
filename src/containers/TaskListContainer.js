import React, {Component} from 'react';
import {connect} from 'react-redux';
import {getTasks,
        deleteTask,
        createTask,
        editFilterBy,
        newProfilePic,
        uploadProfileImage,
        unsetProfilePic,
        toggleTaskUser,
      } from '../actions/taskActions';
import TaskListView from '../components/TaskListView';

//container for AppView, recieves child components from react router
const TaskListContainer = (props) => (
        <TaskListView
          tasks={props.tasks}
          createTask={props.createTask}
          deleteTask={props.deleteTask}
          createNewTask={props.createNewTask}
          filterWhos={props.filterWhos}
          filterType={props.filterType}
          filterBy={props.filterBy}
          editFilterBy={props.editFilterBy}
          getTasks={props.getTasks}
          token={props.token}
          userName={props.userName}
          userId={props.userId}
          toggleTaskUser={props.toggleTaskUser}
          taskView={props.taskView}
        />
);

function mapStateToProps(state) {
  return {
    tasks: state.tasks,
    createNewTask: state.createNewTask,
    filterWhos: state.filterWhos,
    filterType: state.filterType,
    filterBy: state.filterBy,
    token: state.token,
    userName: state.userName,
    userId: state.userId,
    taskView: state.taskView

  }
}

function mapDispatchToProps(dispatch) {
  return {
    getTasks: (createdOrEnd, value, whose, token) => dispatch(getTasks(createdOrEnd, value, whose, token)),
    deleteTask: (taskID, token) => dispatch(deleteTask(taskID, token)),
    createTask: (ID, token) => dispatch(createTask(ID, token)),
    editFilterBy: (filterWhos, filterType, date) => dispatch(editFilterBy(filterWhos, filterType, date)),
    toggleTaskUser: () => dispatch(toggleTaskUser())
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TaskListContainer);
