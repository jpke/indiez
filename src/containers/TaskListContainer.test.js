/* eslint-disable no-undef */
/* eslint-disable import/no-named-as-default */

import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from '../store/configureStore';
const store = configureStore();

import TaskListContainer from './TaskListContainer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <TaskListContainer />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
