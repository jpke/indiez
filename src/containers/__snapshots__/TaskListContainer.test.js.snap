exports[`test renders correctly 1`] = `
<div
  id="taskListContainer">
  <section
    id="filterContainer">
    <div
      id="filter">
      <select
        name="selectFilter"
        onChange={[Function]}>
        <option
          value="mine">
          My Tasks
        </option>
        <option
          value="everyone">
          All Tasks
        </option>
      </select>
      <select
        name="selectFilter"
        onChange={[Function]}>
        <option
          value="all">
          All
        </option>
        <option
          value="end">
          End Before
        </option>
        <option
          value="created">
          Created After
        </option>
      </select>
      <input
        defaultValue=""
        id="filterByDate"
        name="filterByDate"
        onChange={[Function]}
        placeholder="yyyy-mm-dd"
        type="date" />
    </div>
    <button
      onClick={[Function]}>
      Filter Tasks
    </button>
    <button
      id="createOrEditButton"
      onClick={[Function]}>
      Create Task
    </button>
  </section>
  <section
    id="tasks">
    <p>
      No Tasks Listed
    </p>
  </section>
</div>
`;
